gammagl/models/hardgat.py

```diff
@@ -0,0 +1,50 @@
+import tensorlayerx as tlx
+from gammagl.layers.conv import HardGATConv
+
+
+class HardGATModel(tlx.nn.Module):
+    r"""The graph attentional operator from the `"Graph Attention Networks"
+    <https://arxiv.org/abs/1710.10903>`_ paper
+
+    Parameters
+    ----------
+        feature_dim (int): input feature dimension
+        hidden_dim (int): hidden dimension
+        num_class (int): number of classes
+        heads (int): number of attention heads
+        drop_rate (float): dropout rate
+        name (str): model name
+
+    """
+
+    def __init__(self, feature_dim,
+                 hidden_dim,
+                 num_class,
+                 heads,
+                 drop_rate,
+                 k,
+                 name=None):
+        super().__init__(name=name)
+
+        self.conv1 = HardGATConv(in_channels=feature_dim,
+                                 out_channels=hidden_dim,
+                                 heads=heads,
+                                 dropout_rate=drop_rate,
+                                 k=k,
+                                 concat=True)
+        self.conv2 = HardGATConv(in_channels=hidden_dim * heads,
+                                 out_channels=num_class,
+                                 heads=heads,
+                                 dropout_rate=drop_rate,
+                                 k=k,
+                                 concat=False)
+        self.elu = tlx.layers.ELU()
+        self.dropout = tlx.layers.Dropout(drop_rate)
+
```

Comment:
I think we should allow users to choose the number of layers instead of fix the number of layers to 2. You may refer to the new pull request: https://github.com/BUPT-GAMMA/GammaGL/pull/101/files#.